src/strategies/DSL/actions/limitSwapAction.ts

```diff
@@ -1,17 +1,53 @@
-import { LimitSwapActionArgs, PrimitiveArgs } from '@brinkninja/types'
+import { LimitSwapActionArgs, PrimitiveArgs, PrimitiveParamValue } from '@brinkninja/types'
+import { convertToX96HexPrice, toBigint, toTokenArgs } from '@brink-sdk/internal';
+import { InvalidInputError } from '../InvalidInputError';
+
+const FLAT_PRICE_CURVE_ADDRESS = '0xC509733B8dDdbab9369A96F6F216d6E59DB3900f';
 
 function limitSwapAction ({
+  id,
   owner,
   tokenIn,
   tokenOut,
   tokenInAmount,
-  tokenOutAmount,
-  price
+  tokenOutAmount
 }: LimitSwapActionArgs): PrimitiveArgs[] {
+  if (tokenOutAmount === undefined) {
+    throw new InvalidInputError('tokenOutAmount is required')
+  }
+
+  let tokenInAmountBN: bigint
+  let tokenOutAmountBN: bigint
+  try {
+    tokenInAmountBN = toBigint(tokenInAmount)
+    tokenOutAmountBN = toBigint(tokenOutAmount)
+  } catch (error) {
+    throw new InvalidInputError(`Failed to convert tokenInAmount or tokenOutAmount to bigint: ${error}`)
+  }
+
+  const hexPrice = convertToX96HexPrice(tokenInAmountBN, tokenOutAmountBN)
+  const priceCurveParams = { address: FLAT_PRICE_CURVE_ADDRESS, params: hexPrice }
+  
+  const tokenArgsIn = toTokenArgs(tokenIn)
```

Comment:
nitpick but this could probably be named `tokenInArgs` and `tokenOutArgs`

Reply:
done